be/src/exec/vectorized/hdfs_scanner.cpp

```diff
@@ -115,20 +115,16 @@ Status HdfsScanner::open(RuntimeState* runtime_state) {
     if (_is_open) {
         return Status::OK();
     }
-#ifndef BE_TEST
-    if (_scanner_params.fs_handle_type == HdfsFsHandle::Type::HDFS) {
-        auto* file = down_cast<HdfsRandomAccessFile*>(_scanner_params.fs.get());
-        RETURN_IF_ERROR(file->open());
-    }
-#endif
+    CHECK(_file == nullptr) << "File has already been opened";
+    ASSIGN_OR_RETURN(_file, _scanner_params.env->new_random_access_file(_scanner_params.path));
     _build_file_read_param();
     auto status = do_open(runtime_state);
     if (status.ok()) {
         _is_open = true;
-#ifndef BE_TEST
-        (*_scanner_params.open_limit)++;
-#endif
-        LOG(INFO) << "open file success: " << _scanner_params.fs->file_name();
+        if (_scanner_params.open_limit != nullptr) {
+            _scanner_params.open_limit->fetch_add(1, std::memory_order_relaxed);
```

Comment:
memory_order_acq_rel ?

Reply:
Both `memory_order_acq_rel` and `memory_order_relaxed` should be ok here.

Reply:
Agree. relaxed model works here.